internal/backends/sqlite/collection.go

```diff
@@ -91,6 +90,156 @@ func (c *collection) Query(ctx context.Context, params *backends.QueryParams) (*
 	}, nil
 }
 
+// prepareWhereClause adds WHERE clause with filters given in the document.
```

Comment:
What do you think of moving pushdown related logic to a new file `pushdown.go`, it's gonna get big.

Reply:
good idea

Reply:
Oops, I believe I have reverted this change, let me quickly fix that